pkg/odo/cli/init/init.go

```diff
@@ -153,20 +162,21 @@ func (o *InitOptions) Run() (err error) {
 		}
 		starterDownloaded = true
 
-		// in case the starter project contains a devfile, read it again
+		// in case the starter project contains a devfile, read it again but override the metadata name with the one set by the user
 		if _, err = o.clientset.FS.Stat(devfilePath); err == nil {
+			personalizedName := devfileObj.GetMetadataName()
 			devfileObj, _, err = devfile.ParseDevfileAndValidate(parser.ParserArgs{Path: devfilePath, FlattenedDevfile: pointer.BoolPtr(false)})
 			if err != nil {
 				return err
 			}
+			metadata := devfileObj.Data.GetMetadata()
+			metadata.Name = personalizedName
+			devfileObj.Data.SetMetadata(metadata)
```

Comment:
That seems hacky to me. Instead of `PersonalizeName` working on the devfile, I would make it return the new name only, and apply this new name after the starter project is downloaded (and when the starter project has a devfile, I would refresh `devfileObj` with the content of this new devfile)

Reply:
When the starter project has a Devfile, the previous behavior was to refresh `devfileObj`, but in the end, the new name would still need to be applied:
https://github.com/redhat-developer/odo/blob/070b8776c1d84db43d91826f13d3f6d0568e4df9/pkg/odo/cli/init/init.go#L148-L170
But ok, I see your point. I will make the changes suggested.
Also, I wanted to avoid calling `DevfileObj#SetMetadataName(string)`, which makes it not clear that it writes the Devfile to disk.

Reply:
Changes done. Please take a look at this commit: [f9f265d](https://github.com/redhat-developer/odo/pull/5553/commits/f9f265d16ec55d99ca72d1bedb78cc7695e3e222)
Ended up using `DevfileObj#SetMetadataName(string)` with a warning comment (as it was done before), namely because it makes the code concise :)